payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetAddPaymentMethodFragment.kt

```diff
@@ -77,7 +78,7 @@ internal class PaymentSheetAddPaymentMethodFragment(
         // TODO(brnunes-stripe): Refactor to support other payment methods
         (
             childFragmentManager
-                .findFragmentById(R.id.payment_method_fragment_container) as? AddCardFragment<*>
+                .findFragmentById(R.id.payment_method_fragment_container) as? CardDataCollectionFragment<*>
```

Comment:
throw an exception for unsupported types?

Reply:
Updated comment.
This is just to support the `CardDataCollectionFragment` that needs this callback to update the payment selection in the ViewModel. If this happens while another fragment is visible, it should just do nothing. `CardDataCollectionFragment` will be replaced soon then this will be removed.